### ISSUE TITLE
Cannot edit link from anchor preview if toolbar is shown

### ISSUE THREAD

**Issue by Aca-jov** [yabwe/medium-editor#1047]
I have input field with medium editor, where `updateOnEmptySelection: true`

In anchor preview i have `showWhenToolbarIsVisible: true` and it works fine, it shows preview when i hover over added link(and I have visible toolbar).

But, when i click on preview to edit it, it redirects me to the inserted url, instead of opening for edit.

Can you confirm if this is issue, or I'm missing some configuration setting, so it's not working fine?
I investigated it, but I couldn't find any setting..

**Comment by nmielnik** [comment]
I was able to reproduce this with the following settings (modifying example in `static-toolbar.html`):

```js
var editorColTwo = new MediumEditor('#column-two', {
    toolbar: {
        buttons: [/* buttons */],
        static: true,
        align: 'center',
        updateOnEmptySelection: true
    },
    anchorPreview: {
        showWhenToolbarIsVisible: true
    },
    buttonLabels: 'fontawesome'
});
```

I'm trying to dig into what the cause is, thanks @Aca-jov for opening the issue!

**Comment by nmielnik** [comment]
The issue that's exposed here is that the toolbar calls `hidePreview` on the anchor preview extension each time the toolbar is positioned (via `setToolbarPosition`).  You can see the code that does this [here](https://github.com/yabwe/medium-editor/blob/master/src/js/extensions/toolbar.js#L539-L541).

`setToolbarPosition` gets called each time mouseup is triggered on the document as long as the toolbar is present.  So, in the case that the toolbar is present at the same time as the anchor-preview (ie when `showWhenToolbarIsVisible` is true for the anchorPreview options) the toolbar will detect mouseup and tell the anchor preview to hide itself via `hidePreview`.  The `hidePreview` method also clears out the reference to the link that it's showing preview for.  Then, when the click handler is triggered for the anchor-preview, it no longer has a reference to the anchor it needs to edit, so `preventDefault()` is never called and the link is allowed to navigate.

Turns out this isn't related to `updateOnEmptySelection`, all you need is `anchorPreview.showWhenToolbarIsVisible` to reproduce this.

**Comment by nmielnik** [comment]
Was able to find a decent fix for this issue in #1050
